File: Go/Algorithms/34.go
package main

import "sort"

func searchRange(nums []int, target int) []int {
	low := sort.Search(len(nums), func(i int) bool { return nums[i] >= target })
	high := sort.Search(len(nums), func(i int) bool { return nums[i] > (target) })
	if low == high {
		return []int{-1, -1}
	} else {
		return []int{low, high - 1}
	}
}

File: Rust/Algorithms/34.rs
impl Solution {
    pub fn search_range(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let low = nums.partition_point(|&v| v < target) as i32;
        let high = nums.partition_point(|&v| v <= target) as i32;
        
        if low == high {
            vec![-1, -1]
        } else {
            vec![low, high - 1]
        }
    }
}